### PR TITLE
api: logic proposal

### PR DESCRIPTION
**Pull request by thibault-martinez**
This PR proposes and opens a discussion about the API logic design.

# Test Plan:
Not applicable.

### CODE REVIEW

File: ciri/api/BUILD [hunk 1]
```diff
@@ -5,6 +5,8 @@ cc_library(
     srcs = ["api.c"],
     hdrs = ["api.h"],
     deps = [
+        "//cclient/request:requests",
```
Comment (tsvisabo) [inline ciri/api/BUILD:8]: should we move `request/response` to a neutral namespace and use it as a dependency on both cIRI/cclient ?

**Review by tsvisabo — CHANGES_REQUESTED** [review]
LGTM

File: ciri/api/BUILD (hunk 1 above)
Comment (thibault-martinez) [reply ciri/api/BUILD:8]: I think so, and serialization too.
Any suggestion where ?

Comment (tsvisabo) [reply ciri/api/BUILD:8]: `common/network`  ?

**Review by tsvisabo — APPROVED** [review]

File: ciri/api/BUILD (hunk 1 above)
Comment (th0br0) [reply ciri/api/BUILD:8]: For the record: we decided to hold off on moving this for now

File: ciri/api/api.c [hunk 2]
```diff
@@ -7,42 +7,212 @@
...
+  return RC_OK;
+}
+
+static retcode_t get_transactions_to_approve(
+    get_transactions_to_approve_req_t const *const req,
+    get_transactions_to_approve_res_t *const res) {
+  return RC_OK;
+}
+
+static retcode_t attach_to_tangle(attach_to_tangle_req_t const *const req,
```
Comment (th0br0) [inline ciri/api/api.c]: These methods should be able to be called directly.
- tests need this
- a custom IRI node with user code might want to call them without going through the API overhead

**Review by th0br0 — CHANGES_REQUESTED** [review]

File: ciri/api/api.c (hunk 2 above)
Comment (thibault-martinez) [reply ciri/api/api.c]: Alright, I was wondering how to test it.

Comment (thibault-martinez) [reply ciri/api/api.c]: Done.

**Review by th0br0 — APPROVED** [review]

**Pull request merged**